processor/transformprocessor/config.go

```diff
@@ -32,7 +31,17 @@ import (
 type Config struct {
 	config.ProcessorSettings `mapstructure:",squash"`
 
-	ottlconfig.Config `mapstructure:",squash"`
+	OTTLConfig `mapstructure:",squash"`
+}
+
+type OTTLConfig struct {
+	Traces     SignalConfig `mapstructure:"traces"`
+	DataPoints SignalConfig `mapstructure:"datapoints"`
```

Comment:
Lets leave this as metrics for now in the config to reduce the number of breaking changes users have to handle in 0.62.0 and to avoid confusion.  It feels wrong for us to list `DataPoints` but use `Traces` and `Logs` instead of `Spans` and `LogRecords`.  Lets handle this config restructure in #14578.

Reply:
That's fine with me. I've left it as metrics.